src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/OpenTsdbSpec.groovy

```diff
@@ -57,7 +57,29 @@ class OpenTsdbSpec extends HealthCheckSpecification {
                  [[cookieHex: String.format("0x%X", DefaultRule.VERIFICATION_BROADCAST_RULE.cookie)]]].combinations())
     }
 
+    @Unroll("GRPC stats are being logged for metric:#metric, tags:#tags")
+    @Tags([HARDWARE, SMOKE_SWITCHES])
+    def "GRPC stats are being logged"(metric, tags) {
+        expect: "At least 1 result in the past 10 minutes"
+        otsdb.query(10.minutes.ago, metricPrefix + metric, tags).dps.size() > 0
+
+        where:
+        [metric, tags] << (
+                [["switch.packet-in.total-packets", "switch.packet-in.total-packets.dataplane",
+                  "switch.packet-in.no-match.packets", "switch.packet-in.apply-action.packets",
+                  "switch.packet-in.apply-action.packets", "switch.packet-in.invalid-ttl.packets",
+                  "switch.packet-in.action-set.packets", "switch.packet-in.group.packets",
+                  "switch.packet-in.packet-out.packets", "switch.packet-out.total-packets.dataplane",
+                  "switch.packet-out.total-packets.host", "switch.packet-out.eth0-interface-up"],
+                 noviflowSwitches.collect { [switchid: it.switchId.toOtsdFormat()] }].combinations()
+        )
+    }
+
     def getUniqueSwitches() {
         topology.activeSwitches.unique { it.ofVersion }
     }
+
+    def getNoviflowSwitches() {
+        northbound.activeSwitches.findAll { it.description =~ /NW[0-9]+.([0-9].[0-9])/ }.unique { it.hardware }
```

Comment:
you can search for `it.manufacturer == "Noviflow Inc."` if you want noviflow switches. Also, I think it makes sense to pick `it.hardware + it.software` for uniqueness, since we can have same hw model but with different firmware and most likely would like to test them both

Reply:
I can't agree more